saw-core/src/SAWCore/SCTypeCheck.hs

```diff
@@ -254,9 +254,13 @@ prettyTCError e = runReader (helper e) ([], Nothing) where
   helper (BadRecordField n ty) =
       ppWithPos [ return ("Bad record field (" ++ show n ++ ") for type")
                 , ishow ty ]
-  helper (BadRecursorApp r ixs arg) =
+  helper (BadRecursorApp r ixs motive motiveTy) =
       ppWithPos [ return "Type mismatch in recursor application"
-                , ishow (Unshared $ FTermF $ RecursorApp r ixs arg)
+                , ishow (Unshared $ FTermF $ RecursorApp r ixs)
+                , pure "motive:"
+                , ishow motive
+                , pure "motiveTy:"
+                , ishow motiveTy
```

Comment:
Is this a user-facing error message? If so, it might be worth cleaning up the `motive`/`motiveTy` parts of the error message, since these mostly reflect the names of things in the implementation.

Reply:
This was removed later.